### ISSUE TITLE
terraform sometimes hangs during "refreshing state"

### ISSUE THREAD

**Issue by jeroenjacobs1205** [terraform-providers/terraform-provider-vsphere#101]
Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### Terraform Version
0.9.8
### Affected Resource(s)
- vsphere_virtual_machine

### Terraform Configuration Files
```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```
our config files are generated dynamically, but it's not a config issue, so I don't think the config files are relevant.

### Debug Output
https://gist.github.com/jeroenjacobs1205/1038648c40687149dc378c177cdea5c1

### Panic Output
none. it just hangs.

### Expected Behavior
terraform should do the specified task

### Actual Behavior
terraform hangs on a random instance when it's refrshing it's state before applying any changes. this issue occurs at a random instance, so hard to reproduce. it happens when you do "terraform plan", "terraform apply", "terraform destroy". 

So the issue occurs during the fresh-phase, before the actual specified action is executed.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
Since it's random it's hard to reproduce. The only thing that might be relevant, is that the instance where it hangs on in the debug output, is a kubernetes-worker and so has different ip addresses (ip address of the host, ip addresses belonging to different pods, K8s service ip's). So maybe this confuses terraform?

### Important Factoids
- vsphere 6.5

**Comment by vancluever** [comment]
Hey @jeroenjacobs1205, sorry you are having this issue and sorry for taking so long to get back to you!

I understand that configuration may not necessarily be the issue here, but can you post one of your generated configs - it will help speed up making a repro as we will know exactly what the pain points are. It sounds like maybe you have nailed it down to a specific configuration so maybe even an approximation of one of the instances that is failing?

Thanks!

**Comment by Xat59** [comment]
Hi guys,

I have the same problem using Terraform v0.10.2 .

**How to reproduce**
Use this configuration :
``` 
variable "VMWARE_USER" {}
variable "VMWARE_PWD" {}
variable "VMWARE_VSPHERE_SERVER" {}

provider "vsphere" {
  user           = "${var.VMWARE_USER}"
  password       = "${var.VMWARE_PWD}"
  vsphere_server = "${var.VMWARE_VSPHERE_SERVER}"

  # if you have a self-signed cert
  allow_unverified_ssl = true
}

resource "vsphere_virtual_machine" "serv1" {
  name   = "serv1"
  folder = "_ASUPP"
  vcpu   = 1
  memory = 1024
  domain = "bla"
  datacenter = "bla"
  cluster = "Production_Cluster_01"

  network_interface {
    label = "0210_XX"
    mac_address = "aa:bb:cc:dd:ee:ff"
  }

  disk {
    size        = 10
    name        = "serv1"
    datastore   = "CLUS01_SATA03_LS"
    bootable    = true
    type        = "thin"
  }
}
```

Now run it :

```
$ terraform init
$ terraform plan
$ terraform apply
```

The previous commands work well, VMs are created .

Now, just plan again without any change
```
$ terraform plan
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

vsphere_virtual_machine.serv1: Refreshing state... (ID: _ASUPP/serv1)
```

It hangs at this step, I just have to kill the process.

Thank a lot guys.

**Comment by vancluever** [comment]
Hey @Xat59 et al,

Since no config has been posted in regards to the original issue, I am going to close this one now under the assumption that it's possibly related to not having IP address information during refresh. This is a known issue and is something we are reviewing changing the behaviour of in future releases.

The templateless VM power on issue I will be tracking in #59, so check there for updates for that issue.

Thanks!

**Comment by yaso195** [comment]
Hey @vancluever, in which release, "not having IP address during refresh" issue will be resolved? Is there any link to track that issue?

**Comment by Xat59** [comment]
I'm agree with @yaso195 , that could be nice to track the issue.

Thank a lot guys !

**Comment by vancluever** [comment]
@yaso195 and @Xat59, this was fixed in [0.3.0](https://github.com/terraform-providers/terraform-provider-vsphere/blob/master/CHANGELOG.md#030-september-14-2017) and has been available since that release. For information on the current behaviour of our waiters please see [this](https://www.terraform.io/docs/providers/vsphere/r/virtual_machine.html#customization-and-network-waiters) section in the VM resource docs.

As far as tracking updates goes, we don't necessarily link issues all of the time or mark issues as duplicates, not saying that this won't happen in the future, but always the best way to check is to search the issue tracker for similar issues and to look at the changelog to see if your specific issue is addressed.

Thanks!
